### PR TITLE
Improve code coverage in `...Formatter`s

### PR DESCRIPTION
**Pull request by ITaluone**
### `TaskFormatter`
Removed unreachable path.
I think we can rely on the return value of `CanHandle()` that we are dealing with a `Task`.
I tried to get into the `Format` method using `Formatter.Format((Task)null)`, but I always ended up in the `NullFormatter`.

### `GuidValueFormatter` / `StringValueFormatter`
An interpolated string has no missed branches for whatever reason.

### `FormatterObjectGraph`
Removed an unused method, which was introduced in https://github.com/fluentassertions/fluentassertions/pull/2144

### `DateTimeOffsetValueFormatter`
Since we first check in `HasTime` that the value also could have milli seconds, the `if` branch can never be entered

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [ ] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [ ] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

File: Src/FluentAssertions/Formatting/TaskFormatter.cs [hunk 1]
```diff
@@ -15,14 +15,10 @@ public bool CanHandle(object value)
...
-        if (value is Task task)
-        {
-            formatChild("type", task.GetType(), formattedGraph);
-            formattedGraph.AddFragment($" {{Status={task.Status}}}");
-        }
-        else
-        {
-            formattedGraph.AddFragment("<null>");
-        }
+        var task = value as Task;
```
Comment (jnyrup) [inline Src/FluentAssertions/Formatting/TaskFormatter.cs:18]: Let's use a hard cast. That should also remove the need for silencing ReSharper

**Review by dennisdoomen — APPROVED** [review]

File: Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs [hunk 2]
```diff
@@ -92,14 +92,7 @@ public void Format(object value, FormattedObjectGraph formattedGraph, Formatting
 
         if (!hasDate && !hasTime)
         {
-            if (HasMilliSeconds(dateTimeOffset))
-            {
-                formattedGraph.AddFragment("0001-01-01 00:00:00." + dateTimeOffset.ToString("fff", CultureInfo.InvariantCulture));
-            }
-            else
-            {
-                formattedGraph.AddFragment("0001-01-01 00:00:00.000");
```
Comment (ITaluone) [inline Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs:101]: You _most definitely_ cannot reach this `if` body, if you have milliseconds set.
Thus, checking inside this `if` whether you have milliseconds set is a bit redundant..

File: Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs [hunk 3]
```diff
@@ -92,14 +92,7 @@ public void Format(object value, FormattedObjectGraph formattedGraph, Formatting
...
-            }
+            formattedGraph.AddFragment("0001-01-01 00:00:00.000");
         }
```
Comment (ITaluone) [inline Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs:96]: You most definitely cannot reach this if body, if you have milliseconds set.
Thus, checking inside this if whether you have milliseconds set is a bit redundant..

File: Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs [hunk 4]
```diff
@@ -92,14 +92,7 @@ public void Format(object value, FormattedObjectGraph formattedGraph, Formatting
...
         if (!hasDate && !hasTime)
         {
-            if (HasMilliSeconds(dateTimeOffset))
-            {
-                formattedGraph.AddFragment("0001-01-01 00:00:00." + dateTimeOffset.ToString("fff", CultureInfo.InvariantCulture));
-            }
-            else
-            {
-                formattedGraph.AddFragment("0001-01-01 00:00:00.000");
-            }
```
Comment (ITaluone) [inline Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs:102]: You most definitely cannot reach this if body, if you have milliseconds set.
Thus, checking inside this if whether you have milliseconds set is a bit redundant..

Comment (IT-VBFK) [reply Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs:102]: In short `!hasTime == true` when `dateTimeOffset` has all zeros in the time parts

File: Src/FluentAssertions/Formatting/FormattedObjectGraph.cs [hunk 5]
```diff
@@ -101,7 +101,7 @@ private void FlushCurrentLine()
     {
         if (lineBuilder.Length > 0)
         {
-            AppendSafely(lineBuilderWhitespace + lineBuilder);
+            AppendSafely($"{lineBuilderWhitespace}{lineBuilder}");
```
Comment (jnyrup) [inline Src/FluentAssertions/Formatting/FormattedObjectGraph.cs:104]: Why start using string interpolation?

Comment (IT-VBFK) [reply Src/FluentAssertions/Formatting/FormattedObjectGraph.cs:104]: Because we don't have any missed coverage branches anymore

Comment (dennisdoomen) [reply Src/FluentAssertions/Formatting/FormattedObjectGraph.cs:104]: It seems that the [general consensus](https://stackoverflow.com/a/72928810/253961) is that string interpolation is more efficient than simple concatenation.

**Review by dennisdoomen — APPROVED** [review]

File: Src/FluentAssertions/Formatting/FormattedObjectGraph.cs (hunk 5 above)
Comment (jnyrup) [reply Src/FluentAssertions/Formatting/FormattedObjectGraph.cs:104]: Performance doesn't care about consensus 😉

I didn't immediately spot that using string interpolation moves the null check of `lineBuilder` inside `DefaultInterpolatedStringHandler.AppendFormatted`, which increases the code coverage.
After understanding that, I'm fine with this change.

```cs
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using System.Text;

BenchmarkRunner.Run<StringConcat>();

[ReturnValueValidator]
[MemoryDiagnoser]
public class StringConcat
{
    private readonly StringBuilder lineBuilder = new("foo");

    [Params("    ")]
    public string LineBuilderWhitespace { get; set; }

    [Benchmark(Baseline = true)]
    public string Concat() => LineBuilderWhitespace + lineBuilder;

    [Benchmark]
    public string Interpolate() => $"{LineBuilderWhitespace}{lineBuilder}";

    [Benchmark]
    public string Opt() => string.Create(LineBuilderWhitespace.Length + lineBuilder.Length, (LineBuilderWhitespace, lineBuilder), (c, state) =>
    {
        state.LineBuilderWhitespace.CopyTo(c);
        state.lineBuilder.CopyTo(0, c[state.LineBuilderWhitespace.Length..], state.lineBuilder.Length);
    });
}
```

| Method      | Mean     | Error    | StdDev   | Ratio | Gen0   | Allocated | Alloc Ratio |
|------------ |---------:|---------:|---------:|------:|-------:|----------:|------------:|
| Concat      | 15.71 ns | 0.055 ns | 0.046 ns |  1.00 | 0.0115 |      72 B |        1.00 |
| Interpolate | 47.03 ns | 0.149 ns | 0.132 ns |  2.99 | 0.0114 |      72 B |        1.00 |
| Opt         | 14.83 ns | 0.185 ns | 0.164 ns |  0.94 | 0.0063 |      40 B |        0.56 |

**Pull request merged**